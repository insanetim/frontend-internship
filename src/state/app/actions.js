import * as types from './types'

export const loadingOn = () => ({
  type: types.LOADING_ON
})

export const loadingOff = () => ({
  type: types.LOADING_OFF
})

export const showNotification = payload => ({
  type: types.SHOW_NOTIFICATION,
  payload
})

export const showModal = payload => ({
  type: types.SHOW_MODAL,
  payload
})

export const hideModal = () => ({
  type: types.HIDE_MODAL
})
