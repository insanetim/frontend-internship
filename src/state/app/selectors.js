export const loadingSelector = state => state.app.loading
export const modalTypeSelector = state => state.app.modalType
export const modalPropsSelector = state => state.app.modalProps
