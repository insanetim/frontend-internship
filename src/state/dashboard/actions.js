import * as types from './types'

export const requestTrending = payload => ({
  type: types.REQUEST_TRENDING,
  payload
})

export const setTrending = payload => ({
  type: types.SET_TRENDING,
  payload
})

export const requestSearch = payload => ({
  type: types.REQUEST_SEARCH,
  payload
})

export const setSearch = payload => ({
  type: types.SET_SEARCH,
  payload
})

export const setSearchQuery = payload => ({
  type: types.SET_SEARCH_QUERY,
  payload
})

export const clearSearch = () => ({
  type: types.CLEAR_SEARCH
})
