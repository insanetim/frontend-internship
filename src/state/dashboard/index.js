import dashboardReducer from './reducers'

export default dashboardReducer
