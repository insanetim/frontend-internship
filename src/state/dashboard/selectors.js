import * as R from 'ramda'

export const trendingSelector = R.path(['dashboard', 'trending'])
export const searchSelector = R.path(['dashboard', 'search', 'search'])
export const searchQuerySelector = R.path(['dashboard', 'search', 'searchQuery'])
