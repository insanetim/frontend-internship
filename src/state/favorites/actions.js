import * as types from './types'

export const fetchFavorites = (payload = 1) => ({
  type: types.FETCH_FAVORITES,
  payload
})

export const setFavorites = payload => ({
  type: types.SET_FAVORITES,
  payload
})
