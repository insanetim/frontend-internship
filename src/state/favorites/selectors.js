import * as R from 'ramda'

export const favoritesSelector = R.path(['favorites'])
