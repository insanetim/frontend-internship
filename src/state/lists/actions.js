import * as types from './types'

export const requestLists = (payload, cb) => ({
  type: types.REQUEST_LISTS,
  payload,
  cb
})

export const setLists = payload => ({
  type: types.SET_LISTS,
  payload
})

export const requestList = (payload, cb) => ({
  type: types.REQUEST_LIST,
  payload,
  cb
})

export const setList = payload => ({
  type: types.SET_LIST,
  payload
})

export const createList = (payload, cb) => ({
  type: types.CREATE_LIST,
  payload,
  cb
})

export const addToList = payload => ({
  type: types.ADD_TO_LIST,
  payload
})

export const removeFromList = payload => ({
  type: types.REMOVE_FROM_LIST,
  payload
})

export const deleteList = (payload, cb) => ({
  type: types.DELETE_LIST,
  payload,
  cb
})
