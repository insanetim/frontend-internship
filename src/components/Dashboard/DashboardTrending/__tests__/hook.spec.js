import { act, renderHook } from '@testing-library/react-hooks'

import { dispatch } from 'src/__mocks__/react-redux'
import { trendingSelector } from 'src/state/dashboard/selectors'
import { fetchTrending } from 'src/state/dashboard/actions'
import useContainer from '../hook'

jest.mock('src/state/dashboard/selectors', () => ({
  trendingSelector: jest.fn(() => ({}))
}))

describe('DashboardSearchResult useContainer hook', () => {
  let result = null

  beforeEach(() => {
    ;({ result } = renderHook(useContainer))

    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot()
  })

  it('checks `handlePagination` method', () => {
    act(() => {
      result.current.handlePagination(3)
    })

    expect(dispatch).toHaveBeenCalledWith(fetchTrending(3))
  })

  it('checks `useEffect` method', () => {
    trendingSelector.mockReturnValue({ data: 'test/data' })
    ;({ result } = renderHook(useContainer))

    expect(dispatch).not.toHaveBeenCalled()
  })
})
