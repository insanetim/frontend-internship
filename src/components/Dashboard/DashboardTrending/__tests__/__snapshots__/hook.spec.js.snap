// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardSearchResult useContainer hook matches snapshot 1`] = `
{
  "handlePagination": [Function],
  "movies": {},
}
`;
